DDG4/python/DDSim/DD4hepSimulation.py

```diff
@@ -428,6 +428,13 @@ def run(self):
         else:
           gen = DDG4.GeneratorAction(kernel, "Geant4InputAction/hepmc%d" % index)
           gen.Input = "Geant4EventReaderHepMC|" + inputFile
+      elif inputFile.endswith(".hepmc3"):
+        if self.hepmc3.useHepMC3:
+          gen = DDG4.GeneratorAction(kernel, "Geant4InputAction/hepmc%d" % index)
+          gen.Parameters = self.hepmc3.getParameters()
+          gen.Input = "HEPMC3FileReader|" + inputFile
+        else:
+          raise RuntimeError("HepMC3 input file support not enabled")
```

Comment:
I would propose to implement the hepmc3 ending check to
https://github.com/AIDASoft/DD4hep/blob/0515488c0f98f3c1a25f5d36a8b9547ad35b6a77/DDG4/python/DDSim/DD4hepSimulation.py#L546-L554
This would give an error earlier and is less likely to result in a long confusing traceback from undoing some python/c++ interplay that might already have happened here?

And then one can just change line 423 to
```python
if inputFile.endswith((".hepmc", ".hepmc3")):
```

Reply:
I don't think combining in L423 is the right place since for hepmc3 files we cannot fall through to Geant4EventReaderHepMC which only reads hepmc version 2.

Reply:
The `.hepmc3` check for `self.hepmc3.useHepMC3 == True` is already done in `__checkFileFormat`, there is no fall through.

Reply:
How is that possible? `self.hepmc3` is set up in the `__init__` ?!?
Or do you mean if useHepMC3 is changed in the steering file or elsewhere? (Which I don't know why I allowed that in the first place in this case...)

Reply:
> How is that possible?

`self.hepmc3` is set up in `__init__`, but is indeed changed in the steering file (potentially). One reason: hepmc(2) files don't need the hepmc3 library and can use their own reader, so in that case you can set `hepmc3.useHepMC3 = false`.

Reply:
But that wouldn't be a problem, would it? If the hepmc2 file is ending with hepmc then the file format check doesn't care if useHepMC3 is on or off, and then in the selection of the input plugin it would matter if useHepMC3 was turned off in the steering file.

Reply:
Maybe I've just completely confused myself. In any case, I committed what I think you mean. This would bail out earlier for hepmc3 files if there is no HepMC3 support. However, if a HepMC3 file is passed with hepmc extension, it passes the early check and will only get caught later. The hepmc3 extension is more the exception than the rule, I find.

Reply:
As far as I can tell, this thread is now resolved and I will mark as such.